packages/riipen-ui/src/components/Radio.jsx

```diff
@@ -40,33 +41,59 @@ class Radio extends React.Component {
     disabled: false
   };
 
+  getLinkedStyles = () => {
+    const theme = this.context;
+
+    return css.resolve`
+      input:checked ~ p,
+      input:disabled ~ p {
+        color: ${theme.palette.grey[400]};
+      }
+
+      input:checked ~ p.primary {
+        color: ${theme.palette.primary.main};
+      }
+
+      input:checked ~ p.secondary {
+        color: ${theme.palette.secondary.main};
+      }
+    `;
+  };
+
   static contextType = ThemeContext;
 
   render() {
     const { checked, classes, color, disabled, label, ...other } = this.props;
 
     const theme = this.context;
 
+    const linkedStyles = this.getLinkedStyles();
+
     return (
       <div className={clsx(classes)}>
         <label htmlFor={other.id}>
-          <Typography>{label}</Typography>
           <input
+            className={linkedStyles.className}
             checked={checked}
             disabled={disabled}
             type="radio"
             {...other}
           />
+          <Typography classes={[linkedStyles.className, color]}>
+            {label}
+            {linkedStyles.styles}
+          </Typography>
           <span
             className={clsx("checkmark", disabled ? "disabled" : null, color)}
           />
         </label>
         <style jsx>{`
           label {
+            color: ${theme.palette.text.secondary};
             cursor: pointer;
             display: block;
             margin-bottom: 12px;
-            padding-left: 35px;
+            padding-left: 23px;
```

Comment:
Am I the only one who gets super triggered by pixels values that aren't divisible by 5?